include/maths/CMultivariateMultimodalPrior.h

```diff
@@ -386,14 +386,21 @@ class CMultivariateMultimodalPrior : public CMultivariatePrior {
         // where w(i) is its weight we can achieve this by multiplying
         // all weights by some factor f in the range [0, 1].
 
-        if (!this->isForForecasting()) {
-            m_Clusterer->propagateForwardsByTime(time);
-        }
-
+        m_Clusterer->propagateForwardsByTime(time);
         for (const auto& mode : m_Modes) {
             mode.s_Prior->propagateForwardsByTime(time);
         }
 
+        // Remove any mode which is non-informative.
+        while (m_Modes.size() > 1) {
+            auto i = std::find_if(m_Modes.begin(), m_Modes.end(), [](const auto& mode) {
+                return mode.s_Prior->isNonInformative();
+            });
+            if (i == m_Modes.end() || m_Clusterer->remove(i->s_Index) == false) {
+                break;
+            }
+        }
```

Comment:
I think it would be worth adding a comment that `m_Modes` gets modified by the merge callback function that `m_Clusterer->remove()` calls.  Without this knowledge it seems that the `find_if` will always find the same mode.

Reply:
Good suggestion! I added a comment in [3cc7624](https://github.com/elastic/ml-cpp/pull/1551/commits/3cc762409bd1cc4df40328ea01765dc617bfd14c).